src/project-manager.ts

```diff
@@ -316,12 +315,8 @@ export class ProjectManager implements Disposable {
 		if (observable) {
 			return observable;
 		}
-		const parts = url.parse(uri);
-		if (!parts.pathname) {
-			return Observable.throw(new Error(`Invalid URI ${uri}`));
-		}
 		// TypeScript works with file paths, not URIs
-		const filePath = parts.pathname.split('/').map(decodeURIComponent).join('/');
+		const filePath = util.uri2path(uri);
```

Comment:
No, TS service does not require path to be in format of the host OS but we shouldn't use url.parse() in some places while leaving uri2path in others, let's be consistent. I'm OK to use url.parse() if we will be able to access local files after translation (you won't be able to access `/c:/foo/bar` on Windows)